lua/toggleterm.lua

```diff
@@ -50,24 +54,41 @@ local function setup_global_mappings()
   end
 end
 
---Create a new terminal or close beginning from the last opened
+-- Create a new terminal if none are present or closes terminals that are
+-- currently opened, or opens terminals that were previously closed.
 ---@param _ number
 ---@param size number?
 ---@param dir string?
 ---@param direction string?
 local function smart_toggle(_, size, dir, direction)
-  local terminals = terms.get_all()
-  if not ui.find_open_windows() then
-    -- Re-open the last terminal used (if exists) or the first terminal toggled.
-    local term_id = terms.get_last_toggled_id() or terms.get_toggled_id()
-    terms.get_or_create_term(term_id, dir, direction):open(size, direction)
+  local has_open, windows = ui.find_open_windows()
+  if not has_open then
+    local opened = false
+    if not vim.tbl_isempty(terminal_view) then
+      for _, term_id in pairs(terminal_view) do
+        local term = terms.get(term_id)
+        if term then
+          term:open(size, direction)
+          opened = true
+        end
+      end
+      terminal_view = {}
+    end
```

Comment:
Could this logic be factored out into well named helper so it could easily be wrapped in an option, the thinking being if a large enough number of people push back because they liked the old mechanism or they just got used to it etc. it would be nice to be able to configure it on or off but by default my preference is to just change the behaviour without a bunch of options and complexity and see what happens

Reply:
> Could this logic be factored out into well named helper so

sure, I'll do it in a helper.

> but by default my preference is to just change the behaviour without a bunch of options and complexity and see what happens

Exactly my thinking, as this feature seems to me as a sensible default, because it's usually want you want and how other IDEs behave in this regard. I would maybe go with the change and make it non-configurable and then if there are complaints we can make it on/off, as you wish :D

Reply:
done, this is moved to `ui.open_terminal_view` function